libs/@hashintel/design-system/src/block-prompt-input.tsx

```diff
@@ -0,0 +1,223 @@
+          const textWidth = temp.offsetWidth;
+          const textHeight = temp.offsetHeight;
+          document.body.removeChild(temp);
+
+          const firstLineOverflowsButton =
+            width -
+              paddingLeft -
+              paddingRight -
+              textWidth -
+              (hasMultipleLines ? 170 : 0) <
+            0;
+
+          const multipleLines = textHeight / lineHeight > 1;
+
+          setHasMultipleLines(firstLineOverflowsButton || multipleLines);
+        }
+      };
+
+      const submit = (event: FormEvent) => {
+        event.preventDefault();
+        onSubmit?.();
+      };
+
+      return (
+        <Box component="form" onSubmit={submit} sx={{ mb: 0 }}>
+          <TextField
+            {...props}
+            autoFocus
+            multiline
+            onKeyDown={(event) => {
+              const { shiftKey, code } = event;
+              if (!shiftKey && code === "Enter") {
+                onSubmit?.(event);
+              }
+            }}
+            onChange={(event) => {
+              calculateMultipleLines();
+              onChange?.(event);
+            }}
+            required
+            ref={ref}
+            inputRef={inputRef}
+            disabled={disabled}
+            sx={[
+              ({ palette, transitions, spacing }) => ({
+                display: "flex",
+                alignItems: "flex-start",
+                maxWidth: 580,
+                width: 1,
+                [`& .${inputBaseClasses.root}`]: {
+                  width: 1,
+                },
+                [`& .${inputBaseClasses.focused}, .${inputBaseClasses.disabled}`]:
+                  {
+                    boxShadow: "0px 1px 5px rgba(27, 33, 40, 0)",
+                  },
+                [`& .${inputBaseClasses.input}`]: {
+                  minHeight: "unset",
+                  fontSize: 16,
+                  lineHeight: "21px",
+                  paddingY: 2.125,
+                  paddingX: `${spacing(2.75)} !important`,
+                  paddingBottom: hasMultipleLines ? 7.5 : 2.125,
+                  transition: transitions.create("padding-bottom"),
+                },
+                [`& .${inputBaseClasses.disabled}`]: {
+                  background: palette.gray[10],
+                  color: palette.gray[70],
+                },
+                [`& .${outlinedInputClasses.notchedOutline}`]: {
+                  border: `1px solid ${palette.gray[20]}`,
+                  transition: transitions.create("border-color"),
+                },
+                ":hover": {
+                  [`& .${outlinedInputClasses.notchedOutline}`]: {
+                    borderColor: palette.gray[40],
+                  },
+                },
+                [`& .${inputBaseClasses.focused}`]: {
+                  [`& .${outlinedInputClasses.notchedOutline}`]: {
+                    borderWidth: "1px !important",
+                    borderColor: `${palette.blue[70]} !important`,
```

Comment:
See prior comment on whether we can avoid `!important` here

Reply:
Done in 49f5948f67414296033c7d132a86171addf3622d